packages/teleport/src/Apps/AppList/AwsLaunchButton/AwsLaunchButton.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import styled from 'styled-components';
+import { ButtonBorder, Text } from 'design';
+import { Cell } from 'design/DataTable';
+import Menu, { MenuItem } from 'design/Menu';
+import { CarrotDown } from 'design/Icon';
+import cfg from 'teleport/config';
+import { AwsRole } from 'teleport/services/apps';
+
+class AwsLaunchButton extends React.Component<AwsLaunchButtonProps> {
+  anchorEl = React.createRef();
+
+  state = {
+    open: false,
+    anchorEl: null,
+  };
+
+  onOpen = () => {
+    this.setState({ open: true });
+  };
+
+  onClose = () => {
+    this.setState({ open: false });
+  };
+
+  render() {
+    const { open } = this.state;
+    const { awsRoles, fqdn, clusterId, publicAddr } = this.props;
+    return (
+      <Cell align="right">
+        <ButtonBorder
+          width="88px"
+          size="small"
+          setRef={e => (this.anchorEl = e)}
+          onClick={this.onOpen}
+        >
+          LAUNCH
+          <CarrotDown ml={1} fontSize={2} color="text.secondary" />
+        </ButtonBorder>
+        <Menu
+          menuListCss={() => ({
+            maxHeight: '240px',
+            overflowY: 'scroll',
+          })}
+          transformOrigin={{
+            vertical: 'top',
+            horizontal: 'right',
+          }}
+          anchorOrigin={{
+            vertical: 'center',
+            horizontal: 'right',
+          }}
+          getContentAnchorEl={null}
+          anchorEl={this.anchorEl}
+          open={open}
+          onClose={this.onClose}
+        >
+          <RoleItemList
+            awsRoles={awsRoles}
+            fqdn={fqdn}
+            clusterId={clusterId}
+            publicAddr={publicAddr}
+          />
+        </Menu>
+      </Cell>
+    );
+  }
+}
+
+function RoleItemList({
+  awsRoles,
+  fqdn,
+  clusterId,
+  publicAddr,
+}: AwsLaunchButtonProps) {
+  const awsRoleItems = awsRoles.map((item, key) => {
+    const { display, arn } = item;
+    const launchUrl = cfg.getAppLauncherRoute({
+      fqdn,
+      clusterId,
+      publicAddr,
+      arn,
+    });
+    return (
+      <StyledMenuItem
+        key={key}
+        px={2}
+        mx={2}
+        as="a"
+        href={launchUrl}
+        target="_blank"
+      >
+        {display}
+      </StyledMenuItem>
+    );
+  });
+
+  return (
+    <>
+      <Text color="text.onLight" px={2} mx={2} mt={2}>
+        IAM Roles:{' '}
+      </Text>
+      {awsRoleItems}
+    </>
+  );
+}
+
+type AwsLaunchButtonProps = {
+  awsRoles: AwsRole[];
+  fqdn: string;
+  clusterId: string;
+  publicAddr: string;
+};
+
+export default AwsLaunchButton;
+
+const StyledMenuItem = styled(MenuItem)(
```

Comment:
When looking at the recording, i can tell that the margins are off on hover effect. Lets make them  equal size.

![image](https://user-images.githubusercontent.com/7816406/124831995-fae74800-df49-11eb-9c2b-71c9e08c175a.png)


Also, lets have a max-width (40) on the item text and have the text truncated with ellipsis. Make sure that tootltip comes up on hover (`title` attribute)

Change the `IAM Roles:` to:  `Select AIM role`

Reply:
I'm guessing you mean `40ch`, though this seems a bit long to me. How about a max of `25ch`, what do you think?

### max-width: 40ch
![image](https://user-images.githubusercontent.com/56373201/124836866-bb245e80-df51-11eb-97ab-1ca271283c73.png)

### max-width: 25ch
![image](https://user-images.githubusercontent.com/56373201/124836902-cd060180-df51-11eb-8119-c996c3fa2f0f.png)